tests/Helper/QueryHelperTest.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+namespace Szemul\Database\Test\Helper;
+
+use PHPUnit\Framework\TestCase;
+use Szemul\Database\Connection\MysqlConnection;
+use Szemul\Database\Helper\QueryHelper;
+use Szemul\Database\Result\QueryResult;
+
+class QueryHelperTest extends TestCase
+{
+    private MysqlConnection $connection;
```

Comment:
The type should be `MysqlConnection|MockInterface` and most of the errors disappear plus phpstorm and code completion works properly...